assets/js/googlesitekit/data/utils.test.js

```diff
@@ -124,6 +128,231 @@ describe( 'data utils', () => {
+			reducer: combineStoresFakeReducerOne,
+			resolvers: combineStoresFakeResolversOne,
+			selectors: combineStoresFakeSelectorsOne,
+		};
+
+		const storeTwo = {
+			INITIAL_STATE: { two: 2 },
+			actions: combineStoresFakeActionTwo,
+			controls: combineStoresFakeControlsTwo,
+			reducer: combineStoresFakeReducerTwo,
+			resolvers: combineStoresFakeResolversTwo,
+			selectors: combineStoresFakeSelectorsTwo,
+		};
+
+		const storeThree = {
+			INITIAL_STATE: { three: 3 },
+		};
+
+		const storeFour = {
+			reducer: combineStoresFakeReducerFour,
+		};
+
+		const storeFive = {
+			actions: {
+				fakeActionOne() {
+					return { type: 'ACTION_ONE', payload: {} };
+				},
+			},
+		};
+
+		const storeSix = {
+			controls: {
+				FAKE_CONTROL_ONE: () => {
+					return null;
+				},
+			},
+		};
+
+		const storeSeven = {
+			resolvers: {
+				*getFakeActionOne() {
+					yield combineStoresFakeActionOne.fakeActionOne();
+				},
+			},
+		};
+
+		const storeEight = {
+			selectors: {
+				getOne: ( state ) => {
+					return state.one;
+				},
+			},
+		};
+
+		it( 'should combine multiple stores into one', () => {
+			const expectedCombinedStore = {
+				INITIAL_STATE: collect( storeOne.INITIAL_STATE, storeTwo.INITIAL_STATE ),
+				actions: collectActions( storeOne.actions, storeTwo.actions ),
+				controls: collectControls( storeOne.controls, storeTwo.controls ),
+				reducer: collectReducers( storeOne.reducer, storeTwo.reducer ),
+				resolvers: collectResolvers( storeOne.resolvers, storeTwo.resolvers ),
+				selectors: collectSelectors( storeOne.selectors, storeTwo.selectors ),
+			};
+			expect( JSON.stringify( combineStores( storeOne, storeTwo ) ) ).toBe( JSON.stringify( expectedCombinedStore ) );
+		} );
+
+		it( 'should modify combined state', () => {
+			const combinedStore = combineStores( storeOne, storeTwo );
+			let state = combinedStore.reducer();
+
+			expect( state ).toEqual( { one: 1, two: 2 } );
+
+			// It should still respond to the original actions.
+			state = combinedStore.reducer( state, combinedStore.actions.fakeActionOne() );
+			expect( state ).toEqual( { one: true, two: 2 } );
+
+			state = combinedStore.reducer( state, combinedStore.actions.fakeActionTwo() );
+			expect( state ).toEqual( { one: true, two: 2 } );
+		} );
+
+		it( 'should combine stores missing any of the keys', () => {
+			const combinedStoreMissing = {
+				INITIAL_STATE: collect( storeOne.INITIAL_STATE, storeTwo.INITIAL_STATE, storeThree.INITIAL_STATE ),
+				actions: collectActions( storeOne.actions, storeTwo.actions ),
+				controls: collectControls( storeOne.controls, storeTwo.controls ),
+				reducer: collectReducers( storeOne.reducer, storeTwo.reducer ),
+				resolvers: collectResolvers( storeOne.resolvers, storeTwo.resolvers ),
+				selectors: collectSelectors( storeOne.selectors, storeTwo.selectors ),
+			};
+			expect( JSON.stringify( combineStores( storeOne, storeTwo, storeThree ) ) ).toBe( JSON.stringify( combinedStoreMissing ) );
```

Comment:
I don't think this test is quite right. Stringifying these stores changes them in pretty fundamental ways: the actions, controls, reducers, and selectors all contain functions, which may not stringify accurately. Additionally: the aim here is to ensure two things:

1. The combined store has all of the actions, initial state, etc. of each individual store.
2. The utility doesn't error when certain pieces of a store (eg. controls, or even initial state) are _not_ supplied.

It sounds like the name of this test intends to test the second requirement, but it provides every required key. You seem to be thinking of each store as a complete set of items, and omitting parts of the third store, but the utility has no concept of what we consider "the entire store", so this test doesn't seem to be doing anything other than testing what things `stringify` to. I think this one can be removed, unless I'm missing something 😅

Reply:
Makes sense, I've changed tactics, instead of testing the whole object (which required the stringification), I've switched to specifically testing the combined stores for the presence of each individual action, control.

Also this test has been removed, in favor of testing for the combination of several stores which each contain only one of the keys, then testing to makes sure they work together as expected once combined.

Let me know if that all makes sense, thanks!